internal/commands/tui.go

```diff
@@ -383,6 +390,132 @@ func ItemToTUIItem(i store.Item) TUIItem {
 	}
 }
 
+// Struct to aid in filtering items into ranks for BubbleTea
+type Filterer struct {
+	FeedNames []string
+	Term      struct {
+		Title     string
+		FeedNames []string
+	}
+}
+
+// Filters by specific filterValue/s on the Filterer.Term
+func (f *Filterer) FilterBy(filterValues []string, targetFilterValues []string, ranks []fuzzy.Match) []fuzzy.Match {
+	if filterValues != nil && len(filterValues) > 0 {
+		var filteredRanks []fuzzy.Match
+		for _, filterValue := range filterValues {
+			for _, rank := range ranks {
+				if strings.ToLower(targetFilterValues[rank.Index]) == filterValue {
+					filteredRanks = append(filteredRanks, rank)
+				}
+			}
+		}
+		return filteredRanks
+	}
+
+	return ranks
+}
+
+// Breaks what's returned from TUIItem.FilterValue() into a TUIItem.
+func (f *Filterer) GetItem(filterValue string) TUIItem {
+	var i TUIItem
+
+	splits := strings.Split(filterValue, "||")
+
+	i.Title = splits[0]
+	i.FeedName = strings.ToLower(splits[1])
+
+	return i
+}
+
+// Extracts `tag:.*` from the stored f.Term.Title
+func (f *Filterer) ExtractFiltersFor(tag string) []string {
+	var tags []string
+	done := false
+	for done == false {
+		complete := regexp.MustCompile(tag + `:(([^"'][^ ]+)|"([^"]+)"|'([^']+)')`)
+		incomplete := regexp.MustCompile(tag + `:("[^"]*|'[^']*)`)
```

Comment:
I suppose I should explain. `complete` is essentially 3 potential captures separated by `|`, using `[^"]` as a character that *isn't* a `"`, `[^']` for a character that *isn't* a `'`, etc. So `complete` was meant to match `feed:something` (first capture), `feed:"something with spaces"` (second capture), and `feed:'something with single quotes'`.

`incomplete` is a bit more controversial I would say, as it basically pulls out tags that aren't finished from the search, like `feed:"something without and end quote`.

Regex might not be the right solution in general, and this might need some tweaking / clarification. Honestly, this is likely the part that needs the *most* testing / tweaking (because *ew*, regex haha).

Reply:
Thanks for explaining, the gif was mostly in jest at the regex-foo. I think there would be value in adding some of your reply as a comment in the code this for future readers to understand the why behind it